internal/signal/signal.go

```diff
@@ -43,8 +43,6 @@ func newHaltSignalChan() chan os.Signal {
 func StopOnExitSignal(s Stopper) {
 	killSignals := newHaltSignalChan()
 
-	go func() {
-		<-killSignals
-		s.Stop()
-	}()
+	<-killSignals
+	s.Stop()
```

Comment:
Hmm... why this change? If I understood the old code correctly, we intentionally wanted an async stopping channel processing.